### ISSUE TITLE
MyPy error with "show" argument to reswap: incompatible type "str"

### ISSUE THREAD

**Issue by danjac** [adamchainz/django-htmx#421]
### Python Version

3.12.1

### Django Version

5.0.2

### Package Version

1.17.2

### Description

The utility function `django_htmx.http.reswap` appears to take a `Literal` of specific strings, e.g. `innerHTML` or `outerHTML`.

However, HTMX has a number of other options for the `hx-swap` directive: https://htmx.org/attributes/hx-swap/. For example, you can have something like: 

```html
hx-swap="outerHTML show:window:top"
```

In my case, I have code like this:

```python
reswap(response, "outerHTML show:#target_id:top")
```

While this works perfectly well as expected, when doing type checking with MyPY (version 1.8.0) I get the following error:

```
error: Argument 2 to "reswap" has incompatible type "str"; expected "Literal['innerHTML', 'outerHTML', 'beforebegin', 'afterbegin', 'beforeend', 'afterend', 'delete', 'none']" 
```

Of course, the workaround is to use the `#type:ignore` comment, but it seems that `reswap` should not be using a Literal here, as there are any number of possible `hx-swap` values?

**Comment by adamchainz** [comment]
The modifiers didn't exist when the type hint was written. Let's replace it with str now. Please open a PR incoming a changelog note.

**Comment by danjac** [comment]
I was thinking that the literals are quite useful, so instead of getting rid of them perhaps we can add another argument? Something like:

```python

reswap("outerHTML", modifier="show:window:top")
```

**Comment by adamchainz** [comment]
I'd rather not diverge from the syntax in htmx, and there might be more changes in the future.

**Comment by adamchainz** [comment]
Fixed in #422.
